cmd/opampsupervisor/supervisor/supervisor.go

```diff
@@ -460,6 +465,48 @@ func (s *Supervisor) loadAgentEffectiveConfig() {
 	}
 
 	s.effectiveConfig.Store(string(effectiveConfigBytes))
+
+	if s.config.Capabilities.AcceptsRemoteConfig != nil &&
+		*s.config.Capabilities.AcceptsRemoteConfig &&
+		s.config.Storage != nil {
+		// Try to load the last received remote config if it exists.
+		lastRecvRemoteConfig, err := os.ReadFile(filepath.Join(s.config.Storage.Directory, lastRecvRemoteConfigFile))
```

Comment:
I realize it would make the implementation a little more complex, but what are your thoughts on individually persisting each config file passed in the `RemoteConfig` message instead of the message itself? That way the files would be available to be easily read by an operator if necessary. Granted, I'm not sure how useful that would be in practice.

I think persisting them separately could also make it easier in the future to separately pass the files to the Collector like `otelcol --config config1.yaml --config config2.yaml [...]`. This would let the Collector do the merging instead of the Supervisor doing it.

Reply:
That's a good point, I agree.

Reply:
Can we do this after addressing https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30622? I started working on the change and soon ended up changing the effective config report part as well which makes this change more complex.

Reply:
That's good with me.